src/Simplic.FileStructure.Workflow.Data.DB/DocumentWorkflowAssignmentRepository.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Dapper;
+using Simplic.Cache;
+using Simplic.Data.Sql;
+using Simplic.Sql;
+
+namespace Simplic.FileStructure.Workflow.Data.DB
+{
+    /// <summary>
+    /// Represents the document workflow assignment repository 
+    /// </summary>
+    public class DocumentWorkflowAssignmentRepository : SqlRepositoryBase<Guid, DocumentWorkflowAssignment>, IDocumentWorkflowAssignmentRepository
+    {
+        private readonly ISqlService sqlService;
+        public DocumentWorkflowAssignmentRepository(ISqlService sqlService, ISqlColumnService sqlColumnService, ICacheService cacheService) : base(sqlService, sqlColumnService, cacheService)
+        {
+            UseCache = true;
+            this.sqlService = sqlService;
+        }
+
+        /// <summary>
+        /// Table name in the database
+        /// </summary>
+        public override string TableName => "IT_Document_Workflow_Assignment";
+
+        /// <summary>
+        /// Primary key colunm in the database table 
+        /// </summary>
+        public override string PrimaryKeyColumn => "Guid";
+
+        public bool AlreadyExists(Guid documentId)
```

Comment:
Isn't here the workflow-id missing? Furthermore, maybe name it `IsInWorkflow`

Reply:
Its not missing because we just need to check if the document is the table, therefore it must be assigned to a workflow